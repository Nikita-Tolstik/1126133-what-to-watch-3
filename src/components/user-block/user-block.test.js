import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';
import UserBlock from './user-block.jsx';
import history from '../../history.js';
import configureStore from "redux-mock-store";
import {Provider} from 'react-redux';
import NameSpace from '../../reducer/name-space.js';

const mockStore = configureStore([]);

const NO_USER_INFO = `NO_USER_INFO`;
const USER_INFO = `/wtw/static/avatar/7.jpg`;
const AuthorizationStatus = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`,
};


it(`Render UserBlock - AuthorizationStatus.NO_AUTH`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.NO_AUTH,
      userInfo: NO_USER_INFO,
    }
  });

  const tree = renderer.create(
      <Router
        history={history}
      >
        <Provider store={store}>
          <UserBlock />
        </Provider>
      </Router>
  );

  expect(tree).toMatchSnapshot();
});


it(`Render UserBlock - AuthorizationStatus.AUTH`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.AUTH,
      userInfo: USER_INFO,
    }
  });

  const tree = renderer.create(
      <Router
        history={history}
      >
        <Provider store={store}>
          <UserBlock />
        </Provider>
      </Router>
  );

  expect(tree).toMatchSnapshot();
});
