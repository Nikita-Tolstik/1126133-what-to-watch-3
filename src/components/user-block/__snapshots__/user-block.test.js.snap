// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render UserBlock - AuthorizationStatus.AUTH 1`] = `
<div
  className="user-block"
>
  <a
    href="/mylist"
    onClick={[Function]}
  >
    <div
      className="user-block__avatar"
    >
      <img
        alt="User avatar"
        height="63"
        src="https://htmlacademy-react-3.appspot.com/wtw/static/avatar/7.jpg"
        width="63"
      />
    </div>
  </a>
</div>
`;

exports[`Render UserBlock - AuthorizationStatus.NO_AUTH 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="/login"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
